README.md

```diff
@@ -1,2 +1,95 @@
-# msgraph-sdk-dotnet-auth
-MSAL Authentication Provider for .NET
+# Microsoft Graph .NET Authenttication Provider Library
+Get started with Microsoft Graph .NET Authentication Provider Library by integrating Microsoft Graph API into your .Net application.
+Microsoft Graph .NET Authentication Provider Library targets .NetStandard 1.3 and depends on [Microsoft.Identity.Client 2.7.1](https://www.nuget.org/packages/Microsoft.Identity.Client/2.7.1).
+# Installation via NuGet
+To install the authentication provider library via Nuget:
+- Search for `Microsoft.Graph.Auth` in NuGet or 
+- Type `Install-Package Microsoft.Graph.Auth` into the Package Manager Console.
+# Getting Started
+## 1. Register your application
+Register your application to use Microsoft Graph API using one of the following
+supported authentication portals:
+
+* [Microsoft Application Registration Portal](https://apps.dev.microsoft.com):
+  Register a new application that works with Microsoft Account and/or
+  organizational accounts using the unified V2 Authentication Endpoint.
+* [Microsoft Azure Active Directory](https://portal.azure.com): Register
+  a new application in your tenant's Active Directory to support work or school
+  users for your tenant or multiple tenants.
+
+## 2. Create IAuthenticationProvider object
+### 2.1. Confidential Client Providers
+#### a. Authorization code provider
```

Comment:
I highly suggest that you provide a summary of use cases for each provider. OAuth is a foreign concept to a segment of developers still so let's err on the side of too much information. Provide links out to further information.

Reply:
We should point customers to the documentation for what the different factory method arguments represent